### ISSUE TITLE
sf::Ftp::getDirectoryListing() stuck if there are no files/directories

### ISSUE THREAD

**Issue by StDH** [SFML/SFML#1025]
I did some testing and when i added file/directory it worked immediately.

Client: SFML 2.3.2
Server: FileZilla FTP Server (latest)

Permissions were given all and it was hosted on localhost

**Comment by eXpl0it3r** [comment]
You should provide a problem description. What does "stuck" mean exactly? When does it happen? etc.

You might also want to take a look at the [contribution guidelines](http://www.sfml-dev.org/contribute.php) in the future.

**Comment by StDH** [comment]
This is my first time, and i was expecting someone complaining about my "problem description" /^^\

So then,
Imagine ideal state of the connection and ftp (no problems)

util::log(util::INFO, "Requesting list... ");
sf::Ftp::ListingResponse list_status = m_ftp.getDirectoryListing();

getDirectoryListing() gets stuck once there are no files/directories to list from the server, server showed success of sending NLST to the client but client was literally stuck on that function, just as when you are waiting for the socket to complete operation.

example:
(000002)27.12.2015 23:57:16 - user(127.0.0.1)> NLST
(000002)27.12.2015 23:57:16 - user (127.0.0.1)> 150 Opening data channel for directory listing of "/"
(000002)27.12.2015 23:57:16 - user (127.0.0.1)> 226 Successfully transferred "/"

and then the client is stuck on getDirectoryListing(); , looks like waiting for some operation to complete

**Comment by jcowgill** [comment]
I can reproduce this if I add an call to `sleep(1)` above [this line in `Ftp.cpp`](https://github.com/SFML/SFML/blob/master/src/SFML/Network/Ftp.cpp#L379).

The client hangs here:
```
(gdb) bt
#0  0x00007ffff6e645cd in __libc_recv (fd=3, buf=buf@entry=0x7fffffffd560, n=n@entry=1024, flags=flags@entry=16384)
    at ../sysdeps/unix/sysv/linux/x86_64/recv.c:29
#1  0x00007ffff7bd47d0 in recv (__flags=16384, __n=1024, __buf=0x7fffffffd560, __fd=<optimized out>)
    at /usr/include/x86_64-linux-gnu/bits/socket2.h:44
#2  sf::TcpSocket::receive (this=this@entry=0x7fffffffdd48, data=data@entry=0x7fffffffd560, size=size@entry=1024, received=@0x7fffffffd0d8: 0)
    at /home/james/deb-pkg/libsfml/libsfml/src/SFML/Network/TcpSocket.cpp:278
#3  0x00007ffff7bc98d9 in sf::Ftp::getResponse (this=this@entry=0x7fffffffdd40) at /home/james/deb-pkg/libsfml/libsfml/src/SFML/Network/Ftp.cpp:400
#4  0x00007ffff7bcdbc6 in sf::Ftp::getDirectoryListing (this=0x7fffffffdd40, directory=...)
    at /home/james/deb-pkg/libsfml/libsfml/src/SFML/Network/Ftp.cpp:224
#5  0x0000000000401646 in main () at ftptest.cpp:43
```

I think this is because in a FTP directory listing, two responses are used to mark the start (150) and end (266) of the directory listing. `getResponse` is called twice (once through `sendCommand`) to receive both of these responses, but if both are already in the socket's receive buffer during the first call, the first `getResponse` will consume both and then the second call will hang with nothing to receive.

Testcase: https://gist.github.com/jcowgill/f33b4b839b57b5e14bf0

**Comment by binary1248** [comment]
Should be fixed in #1025.

@StDH, @jcowgill care to test?

**Comment by StDH** [comment]
@binary1248 no problem, maybe today or tomorrow for sure
